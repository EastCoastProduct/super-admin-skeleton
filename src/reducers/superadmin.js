import store from 'store';
import { fromJS } from 'immutable';
import { LOGIN_SUCCESS } from '../constants/actions';

const initialState = () => fromJS(store.get('superadmin')) || fromJS({});

export default (state = initialState(), action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      return state.merge(fromJS(action.superadmin));
    default:
      return state;
  }
};
