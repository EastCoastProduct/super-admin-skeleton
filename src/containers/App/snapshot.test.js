import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { superadmin } from '../../fixtures/superadmin';
import { AppComponent } from './';

describe('App component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(
      <AppComponent
        dispatch={() => {}}
        router={{}}
        superadmin={fromJS(superadmin)}
      >
        <main>Test</main>
      </AppComponent>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
