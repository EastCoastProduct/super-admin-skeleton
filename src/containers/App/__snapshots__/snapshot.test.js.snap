exports[`App component snapshot renders basic required data 1`] = `
<div>
  <header
    className="header_uusume">
    <p>
      Hello,Â 
      <b>
        superadmin
         
        superadmin
      </b>
    </p>
    <a
      className="link_1lrrya9"
      href={true}
      onClick={[Function]}>
      Logout
    </a>
  </header>
  <aside
    className="aside_r524fi">
    <nav>
      <ul>
        <li>
          <a
            className="navLink_1dyv036"
            onClick={[Function]}
            style={Object {}}>
            <i
              className="navIcon_11u5mqt fa fa-user" />
            Users
          </a>
        </li>
        <li>
          <a
            className="navLink_1dyv036"
            onClick={[Function]}
            style={Object {}}>
            <i
              className="navIcon_11u5mqt fa fa-address-book" />
            Placeholder
          </a>
        </li>
        <li>
          <a
            className="navLink_1dyv036"
            onClick={[Function]}
            style={Object {}}>
            <i
              className="navIcon_11u5mqt fa fa-lock" />
            Placeholder
          </a>
        </li>
      </ul>
    </nav>
  </aside>
  <div
    className="content_dqivkm">
    <main>
      Test
    </main>
  </div>
</div>
`;
